### DISCUSSION TITLE
Add support for cataloging agents

### DISCUSSION THREAD

**Discussion by acherla** [StreetLamb/tribe#124] [Ideas]
Add an option for cataloging agents to enable the following capabilities
1. Ability to share/copy/re-use agents across teams
2. Ability for agent creator to update an agent and have it automatically apply to multiple team workflows

Future:
1. Ability for agent creator to control access to an agent for other users to leverage (Comprehensive ACL for sharing and providing access to certain agents based on permissions/roles).  To be implemented after agent catalog .  

Ill take this PR

**Comment by StreetLamb** [comment]
@acherla Good suggestion, thanks for working on this! In the future, I can also foresee sharing and reusing team within another team
